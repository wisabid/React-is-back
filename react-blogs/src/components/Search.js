import React from 'react';

const Search = (props) => {
    return (
        <span>
            <input id="search-bar" type="text" placeholder="Search text" value={props.searchval} onChange={props.searchchange}/>
        </span>
    )
}

export default Search;