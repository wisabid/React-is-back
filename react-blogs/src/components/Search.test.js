import React from 'react';
import Search from './Search';
import { shallow } from 'enzyme';
describe('<Blogs />', () => {
    const changeFn = jest.fn(() => {

    });

    it('<Search /> should render correctly', () => {
       const wrapper = shallow(<Search/>);
       expect(wrapper).toMatchSnapshot();
     })    
     
     it('A function should be called when you start typing in Search text box', () => {
        const wrapper = shallow(<Search searchval="abcd" searchchange={changeFn} />);
        wrapper.find('input#search-bar').simulate('change');
        expect(changeFn).toHaveBeenCalled();
      })   
   
      
   })