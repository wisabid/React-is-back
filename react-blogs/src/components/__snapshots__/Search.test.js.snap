// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Blogs /> <Search /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        id="search-bar"
        placeholder="Search text"
        type="text"
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "id": "search-bar",
        "onChange": undefined,
        "placeholder": "Search text",
        "type": "text",
        "value": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          id="search-bar"
          placeholder="Search text"
          type="text"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "search-bar",
          "onChange": undefined,
          "placeholder": "Search text",
          "type": "text",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
